src/polyfem/solver/AdjointNLProblem.cpp

```diff
@@ -175,6 +297,41 @@ namespace polyfem::solver
 
 	bool AdjointNLProblem::is_step_valid(const Eigen::VectorXd &x0, const Eigen::VectorXd &x1)
 	{
+		Eigen::MatrixXd X, V0, V1, V_smooth;
+		Eigen::MatrixXi F;
+
+		int state_count = 0;
+		for (auto state_ : all_states_)
```

Comment:
This should skip states that are not associated with any shape param

Reply:
done